### DISCUSSION TITLE
Livewire Inconsistent Asset Path

### DISCUSSION THREAD

**Discussion by khumam** [livewire/livewire#8085] [1. Help]
I experienced a few issues. I'm using Livewire v3 with some library details as listed below.

![image](https://github.com/livewire/livewire/assets/43531752/139aab12-8037-4473-997b-522da312824c)

Firstly, I store my Laravel application in a subfolder, so the main URL is https://mydomain.com/xyzapp.
Secondly, I'm using an Ubuntu server to run the server with the help of Apache2.
Thirdly, I'm not using @livewireStyles and @livewireScripts.

Inconsistencies occur. When I open my application, the path to livewire.js is https://mydomain.com/livewire/livewire.js, whereas it should be https://mydomain.com/xyzapp/livewire/livewire.js

Referring to the documentation, I've already set up the route as below, which should work in production.

```php
// web.php

if(env('APP_ENV') === 'production') {
    Livewire::setScriptRoute(function ($handle) {
        return Route::get('/xyzapp/livewire/livewire.js', $handle);
    });

    Livewire::setUpdateRoute(function ($handle) {
        return Route::get('/xyzapp/livewire/update', $handle);
    });
}
```

The problem occurs again when I open the application; the path to livewire.js redirects to https://mydomain.com/xyzapp/livewire/livewire.js, but it's not found. Instead, the correct route is https://mydomain.com/xyzapp/xyzapp/livewire/livewire.js, yes double "xyzapp"

Here is my .env config

```
APP_URL=https://mydomain.com/xyzapp
ASSET_URL=https://mydomain.com/xyzapp
```

But somehow it works. And the issue came again when I updated the composer. Run the composer update, and livewire asset not found

Edit: Here is my apache2 config. I redirect the sub path

```
Alias /xyzapp /home/xyzapp/src/public
<Directory /home/xyzapp/src/public>
        Options Indexes MultiViews FollowSymLinks
        AllowOverride All
        Require all granted
</Directory>
```

**Comment by khumam** [comment]
I solved the issue by adding the asset_url inside the livewire.php config:

```php
'asset_url' => env('APP_URL', 'http://localhost') . '/livewire/livewire.min.js',
```
Additionally, I disabled the setScriptRoute() but still kept setUpdateRoute() on my routes.

**Reply by orgabor** [reply]
@khumam thanks for the solution provided. Same workaround helped in my case. I guess livewire has serious problem to serve the infrastructure from a subroute.

**Reply by gpibarra** [reply]
did you find a solution?

I tried this and it works
`.env`
```
APP_URL_ALIAS=/laravel
APP_URL=http://127.0.0.1${APP_URL_ALIAS}
```

`./config/livewire.php`
```php
    'asset_url_alias' => env('APP_URL_ALIAS', ''),

    'asset_url' => str(env('APP_URL_ALIAS', ''))
        ->whenNotEmpty(function (\Illuminate\Support\Stringable $string) {
            return $string->finish('/');
        })
        ->append(config('app.debug') ? 'livewire/livewire.js' : 'livewire/livewire.min.js')
        ->toString(),
```

`./app/Providers/AppServiceProvider.php`
```php
    public function register(): void
    {
        //
        $this->app->singleton(
            \Livewire\Mechanisms\HandleRequests\HandleRequests::class,
            \App\Support\Livewire\MyHandleRequests::class
        );
    }
```
`./app/Support/Livewire/MyHandleRequests.php`
```php
<?php

namespace App\Support\Livewire;

use Illuminate\Support\Facades\Route;
use Livewire\Mechanisms\HandleRequests\HandleRequests as BaseHandleRequests;

class MyHandleRequests extends BaseHandleRequests
{

    function boot()
    {
        // Only set it if another provider hasn't already set it....
        if (! $this->updateRoute) {
            app(BaseHandleRequests::class)->setUpdateRoute(function ($handle) {
                return Route::post('/livewire/update', $handle)->middleware('web');
            });
        }

        $this->skipRequestPayloadTamperingMiddleware();
    }

    function getUpdateUri()
    {
        return (string) str(parent::getUpdateUri())
            ->start('/')->start(config('livewire.asset_url_alias', '/'))->start('/');
    }
}
```
> [!NOTE]  
> The boot method is overridden only by changing `app($this::class)` vs `app(\Livewire\Mechanisms\HandleRequests\HandleRequests::class)`. It didn't work for me because `app()` actually creates a new instance... originally it should have been `app(self::class)` instead of `app($this::class)`
